server/src/tokens.rs

```diff
@@ -0,0 +1,99 @@
+use crate::types::EdgeToken;
+
+pub fn simplify(tokens: &[EdgeToken]) -> Vec<EdgeToken> {
+    tokens
+        .iter()
+        .filter_map(|token| {
+            tokens.iter().fold(Some(token.clone()), |acc, current| {
+                acc.and_then(|lead| {
+                    if current != &lead && current.subsumes(&lead) {
+                        None
+                    } else {
+                        Some(lead)
+                    }
+                })
+            })
+        })
+        .collect()
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::{
+        tokens::simplify,
+        types::{EdgeToken, TokenType},
+    };
+
+    fn test_token(env: Option<&str>, projects: Vec<&str>) -> EdgeToken {
+        EdgeToken {
+            secret: "the-secret".into(),
+            token_type: Some(TokenType::Client),
+            environment: env.map(|env| env.into()),
+            projects: projects.into_iter().map(|p| p.into()).collect(),
+            expires_at: None,
+            seen_at: None,
+            alias: None,
+        }
+    }
+
+    #[test]
+    fn test_case_1() {
```

Comment:
Better name maybe? I would have no idea why this failed at a glance

Reply:
Tryied to describe them better but I wanted to use `#[test_case]` although I couldn't find a way, so the index indicates we should refactor them later